feat/refactor: maze function changes to help with mapping

RosyMapleMoth opened the pull request:
#### Current State <span><a href="#details">( scroll down )</a></span>
- [x] Final

<h1 id="details">Details about this Pull Request</h1>

#### What's this Pull Request do?

Add a Node getter function to Maze, change the Neighbor Node function to a public function.
#### Any background context you want to provide?

While working with mapping it became apparent these changes would be very helpful/necessary  
#### Changes or additions made:
- [added]
  - Node\* getNode( Vector::Pos pos ); 
- [changed] 
  - Node\* getNeighborNode( Vector::Pos pos , direction dir );

RosyMapleMoth committed:
feat/refactor: maze function changes to help with mapping

add a function to get a node at given position from maze, make
getNeighborNode public

I was unsure what to put in the footer

blhough commented on Maze.cpp:
```diff
@@ -116,6 +116,11 @@ namespace Micromouse
+	// Retruns the Node at the given position.
+	Node* Maze::getNode( Vector::Pos pos )
+	{
+		return maze[ pos.x ][ pos.y ];
```
Careful x and y are private for Vector::Pos so you actually need to call functions to get x,y. 

pos.x() , pos.y()

blhough commented:
Other than that one small thing i line commented on. This is a very nice PR and nice code!

I will merge once fixed. 
Any new commits to this branch are automatically added to the PR

blhough commented:
what IDE or compiler are you using?
Make sure to compile the project before making a PR to ensure everything is working

RosyMapleMoth committed:
fix : change how getNode accuses pos x and y

change getNode to use the proper x and y getter functions Pos provides.
Rather than attempting to directly accuse the private data members

RosyMapleMoth committed:
Revert "fix : change how getNode accuses pos x and y"

This reverts commit 0d9902f602e58a27bcfe9424e43c89dbe6cd586c.

RosyMapleMoth committed:
fix : change how getNode accuses pos

change getNodeto to use the proper x and y getter functions Pos
provides. Rather than attempting to directly accuse the private data
members

The last commit and revert accidently had a few other small unintended
changes

RosyMapleMoth commented:
I'm using visual studio 2015 but I hadn't attempted a compile because it was such a small change. my bad, and it won't happen again!

also I've just fixed the getNode
